### PR TITLE
Try another fix for nifti scroll

### PR DESCRIPTION
**Pull request by CYB3RL1F3**
### Context

Impossible to scroll and use any tool in a nifti...
It's getting boring right now!
It started to fail since version 1.46.0

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

- [x] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] "OS: Mac OS X
- [x] "Node version: 20.0.0
- [x] "Browser: Chrome

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by CYB3RL1F3** [comment]
UP

**Comment by CYB3RL1F3** [comment]
Hello @wayfarer3130  @sedghi could you check it and help me to definitely fix this issue?
I have a nifti with a volumeId which is its URL and a volumeId which is the URL with different things in the query string so it doesn't find the volume and process scroll and action with tools.
It's getting boring, seriously!
I try to fix the sh*t !!
Could you merge or fix the problem please??

**Comment by SachidanandAlle** [comment]
need this fix urgently.. the other version of the same PR is https://github.com/cornerstonejs/cornerstone3D/pull/1140
But @CYB3RL1F3 has provided the better version to compare the volumeID.. 

@pieper / @sedghi please help to get this fix pushed asap.  Hopefully we can pull the latest version of NPM package with this fix into our builds.

**Review by wayfarer3130 — APPROVED** [review]

**Comment by SachidanandAlle** [comment]
@CYB3RL1F3 did u try it on examples.. I am still facing the same issue

```
Cannot destructure property 'uid' of 'volumeActors[imageVolumeIndex]' as it is undefined.
TypeError: Cannot destructure property 'uid' of 'volumeActors[imageVolumeIndex]' as it is undefined.
    at getTargetVolumeAndSpacingInNormalDir (http://localhost:3000/niftiWithTools.js:91918:12)
    at getVolumeSliceRangeInfo (http://localhost:3000/niftiWithTools.js:92296:88)
    at Module.getVolumeViewportScrollInfo (http://localhost:3000/niftiWithTools.js:92345:75)
    at scrollVolume (http://localhost:3000/niftiWithTools.js:151218:66)
    at scroll (http://localhost:3000/niftiWithTools.js:151204:5)
    at StackScrollMouseWheelTool.mouseWheelCallback (http://localhost:3000/niftiWithTools.js:117256:66)
    at customCallbackHandler (http://localhost:3000/niftiWithTools.js:98844:29)
    at triggerEvent (http://localhost:3000/niftiWithTools.js:94165:13)
    at HTMLDivElement.wheelListener (http://localhost:3000/niftiWithTools.js:102344:66)
```

**Comment by SachidanandAlle** [comment]
targetVolumeID: nifti:http://10.117.6.209:36868/example-1.nii.gz?sliceIndex=105&viewPlaneNormal=0,0,-1&focalPoint=-196,-196,105

File: packages/core/src/utilities/getTargetVolumeAndSpacingInNormalDir.ts [hunk 1]
```diff
@@ -66,8 +66,8 @@ export default function getTargetVolumeAndSpacingInNormalDir(
 
   // If a volumeId is defined, set that volume as the target
   if (targetVolumeId) {
-    const imageVolumeIndex = imageVolumes.findIndex(
-      (iv) => iv.volumeId === targetVolumeId
+    const imageVolumeIndex = imageVolumes.findIndex((iv) =>
+      iv.volumeId.includes(targetVolumeId)
```
Comment (SachidanandAlle) [inline packages/core/src/utilities/getTargetVolumeAndSpacingInNormalDir.ts:70]: this should be `targetVolumeId.includes(iv.volumeId)`

**Comment by tangy5** [comment]
I'm having same issue here to load our MAISI image, this is blocking.

**Comment by lambacini** [comment]
What if we load the nifti file from an endpoint ? 
for example http://serverip/get_nifti_image?imageid=123456

In this case, the scroll does not work correctly because targetid and volumeid do not match.
targetId = volumeId:nifti:http://serverip/get_image?image_id=123456
volumeId = nifti:http://serverip/get_image

This only seems to solve your problem, i think a more general approach is needed.

Screenshot from 2024-06-28 17-02-57


@sedghi @wayfarer3130

**Pull request merged**